data-pipeline-flink/user-cache-updater/src/main/scala/org/sunbird/dp/usercache/functions/UserCacheUpdaterFunction.scala

```diff
@@ -152,9 +152,121 @@ class UserCacheUpdaterFunction(config: UserCacheUpdaterConfig)(implicit val mapT
       record.getString("type").toLowerCase match {
         case config.stateKey => result.put(config.stateKey, record.getString("name"))
         case config.districtKey => result.put(config.districtKey, record.getString("name"))
+        case config.blockKey => result.put(config.blockKey, record.getString("name"))
       }
     })
     result
   }
 
+  /**
+    * It Will return the rootOrgId to filter out custodian users and state users
+    * @param metrics
+    * @return custodianRootOrgId
+    */
+  def getCustodianRootOrgId(metrics: Metrics): String = {
```

Comment:
This value needs to be cached if the CustodianOrgId doesn't change. We should not be making so many calls when it returns the same value.

Reply:
Done